tests/testmacro.nim

```diff
@@ -80,3 +80,48 @@ suite "Macro transformations test suite":
     check waitFor(testAwait()) == true
   test "`awaitne` command test":
     check waitFor(testAwaitne()) == true
+
+suite "Exceptions tracking":
+  test "Can raise valid exception":
+    proc test1 {.async.} = raise newException(ValueError, "hey")
+    proc test2 {.async, raises: [ValueError].} = raise newException(ValueError, "hey")
+    proc test3 {.async, raises: [IOError, ValueError].} =
+      if 1 == 2:
+        raise newException(ValueError, "hey")
+      else:
+        raise newException(IOError, "hey")
+
+    proc test4 {.async, raises: [].} = raise newException(Defect, "hey")
+
+  test "Cannot raise invalid exception":
+    check not (compiles do:
```

Comment:
I usually introduce a template called `reject` for test cases like this one:

```nim
template reject(body: untyped) =
  check(not compiles(body))

...

reject:
  proc test3 {.async, raises: [IOError].} = raise newException(ValueError, "hey")

```

Reply:
Neat, thanks! Could be a nice addition to unittest2